library/core/src/num/mod.rs

```diff
@@ -820,6 +820,29 @@ impl u16 {
     uint_impl! { u16, u16, i16, NonZeroU16, 16, 65535, 4, "0xa003", "0x3a", "0x1234", "0x3412", "0x2c48",
     "[0x34, 0x12]", "[0x12, 0x34]", "", "" }
     widening_impl! { u16, u32, 16, unsigned }
+
+    /// Checks if the value is a Unicode surrogate code point, which are disallowed values for [`char`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let low_non_surrogate = 0xA000;
+    /// let low_surrogate = 0xD800;
+    /// let high_surrogate = 0xDC00;
+    /// let high_non_surrogate = 0xE000;
+    ///
+    /// assert!(!low_non_surrogate.is_surrogate());
+    /// assert!(low_surrogate.is_surrogate());
+    /// assert!(high_surrogate.is_surrogate());
+    /// assert!(!high_non_surrogate.is_surrogate());
+    /// ```
+    #[must_use]
+    #[unstable(feature = "is_char_surrogate", issue = "none")]
+    #[rustc_const_unstable(feature = "is_char_surrogate", issue = "none")]
+    #[inline]
+    pub const fn is_char_surrogate(self) -> bool {
```

Comment:
Given all the `_ascii_` methods on `u8`, this seems like a reasonable thing to add.

However, I think it needs a different name.  A `char` can never be a surrogate, so mentioning it seems wrong.

I think the `_ascii_` names and `from_utf8` and such mean that it should mention the encoding it's using, not a datatype.

So how about this?
```suggestion
    pub const fn is_utf16_surrogate(self) -> bool {
```

Reply:
You're right; I was originally thinking of going with `is_unicode_surrogate` but decided it was too long.

My main apprehension with calling it `utf16` surrogate is it implies that it's specific to UTF-16, which it actually isn't; it's specific to Unicode, even though its inclusion in Unicode is specific for UTF-16. I will agree that the name is better than the original, though.